convert-hf-to-gguf.py

```diff
@@ -511,6 +511,16 @@ def write_tensors(self):
 class MPTModel(Model):
     model_arch = gguf.MODEL_ARCH.MPT
 
+    def set_vocab(self):
+        try:
+            self._set_vocab_gpt2()
+        except:
```

Comment:
@bryanSwk Why would `_set_vocab_gpt2()` ever fail? Trying to understand what the `except` clause is doing here, and if we should have it, or qualify it a bit more. An open-ended `except` is breaking the CI linter. We can fix this by changing it to `except Exception:`, but I'd prefer to understand why this branch is here.

Reply:
Very helpful, thank you!

Do you happen to know what kind of exception it will throw in that instance? If not, I can just change it to `except Exception:` and it should still pass lint

Reply:
Sounds good! Feel free to make any suggestions or wording changes on this PR:
https://github.com/ggerganov/llama.cpp/pull/6470

I'm not very familiar with SEA-LION, so I welcome any adjustments you may have that would make things more clear. :)